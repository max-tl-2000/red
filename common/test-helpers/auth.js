/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import moment from 'moment'; // eslint-disable-line red/no-moment
import { expect } from 'chai';

export const verifyTokenExpiration = (decodedToken, expectedTokenExpiration) => {
  const iat = moment.unix(decodedToken.iat);
  const exp = moment.unix(decodedToken.exp);

  const result = exp.diff(iat, expectedTokenExpiration.substring(expectedTokenExpiration.length - 1));
  const expectedValue = expectedTokenExpiration.substring(0, expectedTokenExpiration.length - 1);
  expect(result).to.equal(+expectedValue);
};
