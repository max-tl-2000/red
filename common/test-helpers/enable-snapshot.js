/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import chai from 'chai';
import chaiJestSnapshot from 'chai-jest-snapshot';

const enableSnapshot = () => {
  if (!global.before) return;
  chai.use(chaiJestSnapshot);
  before(() => {
    chaiJestSnapshot.resetSnapshotRegistry();
  });

  beforeEach(function beforeEach() {
    const { currentTest } = this;
    chaiJestSnapshot.setFilename(`${currentTest.file}.chai-snap`);
    chaiJestSnapshot.setTestName(currentTest.fullTitle());
  });
};

enableSnapshot();

export { editUnpredictable } from '../helpers/snapshot-helper';
