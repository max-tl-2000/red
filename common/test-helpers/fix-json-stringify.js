/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

const stringify = JSON.stringify;

before(() => {
  JSON.stringify = obj => {
    const seen = [];

    return stringify(obj, (key, val) => {
      if (typeof val === 'object') {
        if (seen.indexOf(val) >= 0) {
          return;
        }
        seen.push(val);
      }
      return val; // eslint-disable-line
    });
  };
});

after(() => {
  JSON.stringify = stringify;
});
