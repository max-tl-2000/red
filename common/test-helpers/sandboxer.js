/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

let _sinon;
let sandboxes = [];

function create() {
  // this is for when this module is used in karma
  _sinon = _sinon || global.sinon;
  const sandbox = _sinon.sandbox.create();
  sandboxes.push(sandbox);

  return sandbox;
}

function restore() {
  sandboxes.forEach(sandbox => sandbox.restore());
  sandboxes = [];
}

function setSinon(sinon) {
  _sinon = sinon;
}

export { create, restore, setSinon };
