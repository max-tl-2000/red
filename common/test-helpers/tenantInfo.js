/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

export const TEST_TENANT_ID = 'af20764d-1e0d-41a5-be54-3766215f89a4';
