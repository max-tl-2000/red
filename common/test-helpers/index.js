/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';
import { unmountComponentAtNode } from 'react-dom';
import when from './when';
import and from './and';
import * as overrider from './overrider';
import * as sandboxer from './sandboxer';
import sleep from '../helpers/sleep';

sandboxer.setSinon(sinon);

const expect = chai.expect;
const assert = chai.assert;

chai.use(sinonChai);

let div;

if (global.document) {
  beforeEach(() => {
    div = document.createElement('div');
    div.setAttribute('id', 'sandbox');

    document.body.appendChild(div);
  });
}

afterEach(() => {
  if (global.document && div) {
    unmountComponentAtNode(div);
    document.body.removeChild(div);
    div = null;
  }

  overrider.restore();
  sandboxer.restore();
});

const getSandBoxDiv = () => div;

// helper method to trigger a real click event in the browser
// jquery.fn.click is not the same as this as the event need
// to also generate the capture phase
const triggerMouseEvent = (ele, evt) => {
  const mEvent = document.createEvent('MouseEvent');
  // interface
  // event.initMouseEvent(type, canBubble, cancelable, view,
  //                    detail, screenX, screenY, clientX, clientY,
  //                    ctrlKey, altKey, shiftKey, metaKey,
  //                    button, relatedTarget);
  mEvent.initMouseEvent(evt, true, true, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);

  ele.dispatchEvent(mEvent);
};

export { getSandBoxDiv, triggerMouseEvent, overrider, expect, assert, sinon, when, sandboxer, and, sleep };
