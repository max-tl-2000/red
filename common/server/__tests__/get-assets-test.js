/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import { override, restore } from '../../test-helpers/overrider';
const { mockModules } = require('test-helpers/mocker').default(jest);

const expect = global.expect;

describe('get-assets-test', () => {
  beforeEach(() => {
    override(
      process.env,
      {
        USE_ASSETS_PROD_DIST_FOLDER: 'true',
      },
      true /* allow non existend props */,
    );
  });

  afterEach(() => {
    restore();
    jest.resetModules();
  });

  describe('when useDevMode = true and query.local = undefined', () => {
    it('should return the list of files from local and no min extensions for main.js and vendors.js', async () => {
      mockModules({
        '../../helpers/xfs': {
          readJSON: manifest => {
            let res;
            if (manifest === 'main') {
              res = Promise.resolve({
                'main.css': 'main.css',
                'main.js': 'main.js',
              });
            }
            if (manifest === 'vendors') {
              res = Promise.resolve({
                'vendors.js': 'vendors.js',
              });
            }
            return res;
          },
        },
      });
      const { getAssets } = require('../get-assets'); // eslint-disable-line global-require
      const result = await getAssets({
        host: 'demo.local.env.reva.tech',
        query: {},
        useDevMode: true,
        jsManifests: ['vendors', 'main'],
        cssFiles: ['main.css'],
        jsFiles: ['vendors.js', 'main.js'],
      });
      expect(result).toMatchSnapshot();
    });
  });

  describe('when useDevMode = false and one of them contain the local flag', () => {
    it('should return the list of files from local and no min extensions for main.js and vendors.js', async () => {
      mockModules({
        '../../helpers/xfs': {
          readJSON: manifest => {
            let res;
            if (manifest === 'main') {
              res = Promise.resolve({
                'main.css': 'main.css',
                'main.js': 'main.js',
              });
            }
            if (manifest === 'vendors') {
              res = Promise.resolve({
                'vendors.js': 'vendors.js',
              });
            }
            return res;
          },
        },
      });
      const { getAssets } = require('../get-assets'); // eslint-disable-line global-require
      const result = await getAssets({
        host: 'demo.local.env.reva.tech',
        query: {},
        useDevMode: false,
        jsManifests: ['vendors', 'main'],
        cssFiles: [{ name: 'main.css', local: true }],
        jsFiles: ['vendors.js', { name: 'main.js', local: true }],
      });
      expect(result).toMatchSnapshot();
    });
  });

  describe('when useDevMode = true and one of them contain the local flag', () => {
    it('should return the list of files from local and min extensions for main.js and vendors.js', async () => {
      mockModules({
        '../../helpers/xfs': {
          readJSON: manifest => {
            let res;
            if (manifest === 'main') {
              res = Promise.resolve({
                'main.css': 'main.css',
                'main.js': 'main.js',
              });
            }
            if (manifest === 'vendors') {
              res = Promise.resolve({
                'vendors.js': 'vendors.js',
              });
            }
            return res;
          },
        },
      });
      const { getAssets } = require('../get-assets'); // eslint-disable-line global-require
      const result = await getAssets({
        host: 'demo.local.env.reva.tech',
        query: {},
        useDevMode: false,
        jsManifests: ['vendors', 'main'],
        cssFiles: [{ name: 'main.css', local: true }],
        jsFiles: ['vendors.js', { name: 'main.js', local: true }],
      });
      expect(result).toMatchSnapshot();
    });
  });

  describe('when NODE_ENV = production and query.local = undefined', () => {
    it('should return the list of files from cdn, min extensions for main.js and vendors.js and main.css and vendor.css', async () => {
      mockModules({
        '../../helpers/xfs': {
          readJSON: manifest => {
            let res;
            if (manifest === 'main') {
              res = Promise.resolve({
                'main.css': 'main.css',
                'main.js': 'main.js',
              });
            }
            if (manifest === 'vendors') {
              res = Promise.resolve({
                'vendors.js': 'vendors.js',
              });
            }
            return res;
          },
        },
      });

      const { getAssets } = require('../get-assets'); // eslint-disable-line global-require
      const result = await getAssets({
        host: 'demo.local.env.reva.tech',
        query: {},
        useDevMode: false,
        jsManifests: ['main', 'vendors'],
        cssFiles: ['main.css'],
        jsFiles: ['vendors.js', 'main.js'],
      });
      expect(result).toMatchSnapshot();
    });
  });

  describe('when NODE_ENV = production and query.local = true', () => {
    it('should return the list of files from cdn ignoring local=true, min extensions for main.js and vendors.js and main.css', async () => {
      mockModules({
        '../../helpers/xfs': {
          readJSON: manifest => {
            let res;
            if (manifest === 'main') {
              res = Promise.resolve({
                'main.css': 'main.css',
                'main.js': 'main.js',
              });
            }
            if (manifest === 'vendors') {
              res = Promise.resolve({
                'vendors.js': 'vendors.js',
              });
            }
            return res;
          },
        },
      });

      const { getAssets } = require('../get-assets'); // eslint-disable-line global-require
      const result = await getAssets({
        host: 'demo.local.env.reva.tech',
        query: {},
        useDevMode: false,
        jsManifests: ['main', 'vendors'],
        cssFiles: ['main.css'],
        jsFiles: ['vendors.js', 'main.js'],
      });
      expect(result).toMatchSnapshot();
    });
  });

  describe('when useDevMode = true and query.local = false', () => {
    it('should return the list of files from cdn, no min extensions for main.js and vendors.js and no main.css', async () => {
      mockModules({
        '../../helpers/xfs': {
          readJSON: manifest => {
            let res;
            if (manifest === 'main') {
              res = Promise.resolve({
                'main.css': 'main.css',
                'main.js': 'main.js',
              });
            }
            if (manifest === 'vendors') {
              res = Promise.resolve({
                'vendors.js': 'vendors.js',
              });
            }
            return res;
          },
        },
      });

      const { getAssets } = require('../get-assets'); // eslint-disable-line global-require
      const result = await getAssets({
        host: 'demo.local.env.reva.tech',
        query: {},
        useDevMode: true,
        jsManifests: ['main', 'vendors'],
        cssFiles: ['main.css'],
        jsFiles: ['vendors.js', 'main.js'],
      });
      expect(result).toMatchSnapshot();
    });
  });

  describe('when NODE_ENV = production and query.min = false', () => {
    it('should return the list of files from cdn, no min extensions for main.js and vendors.js and main.css and vendor.css', async () => {
      mockModules({
        '../../helpers/xfs': {
          readJSON: manifest => {
            let res;
            if (manifest === 'main') {
              res = Promise.resolve({
                'main.css': 'main.css',
                'main.js': 'main.js',
              });
            }
            if (manifest === 'vendors') {
              res = Promise.resolve({
                'vendors.js': 'vendors.js',
              });
            }
            return res;
          },
        },
      });

      const { getAssets } = require('../get-assets'); // eslint-disable-line global-require
      const result = await getAssets({
        host: 'demo.local.env.reva.tech',
        query: { min: 'false' },
        useDevMode: false,
        jsManifests: ['main', 'vendors'],
        cssFiles: ['main.css'],
        jsFiles: ['vendors.js', 'main.js'],
      });
      expect(result).toMatchSnapshot();
    });
  });

  describe('when NODE_ENV = production and query.dMode = true', () => {
    describe('and host is production host', () => {
      it('should return the assets from local', async () => {
        mockModules({
          '../../helpers/xfs': {
            readJSON: manifest => {
              let res;
              if (manifest === 'main') {
                res = Promise.resolve({
                  'main.css': 'main.css',
                  'main.js': 'main.js',
                });
              }
              if (manifest === 'vendors') {
                res = Promise.resolve({
                  'vendors.js': 'vendors.js',
                });
              }
              return res;
            },
          },
        });

        const { getAssets } = require('../get-assets'); // eslint-disable-line global-require
        const result = await getAssets({
          host: 'demo.reva.tech',
          query: { dMode: 'true' },
          useDevMode: false,
          jsManifests: ['main', 'vendors'],
          cssFiles: ['main.css'],
          jsFiles: ['vendors.js', 'main.js'],
        });
        expect(result).toMatchSnapshot();
      });
    });

    describe('and host is production host and query.min = false ', () => {
      it('should return the assets from local and should return no min extensions for main.js and vendors.js and main.css and vendor.css ', async () => {
        mockModules({
          '../../helpers/xfs': {
            readJSON: manifest => {
              let res;
              if (manifest === 'main') {
                res = Promise.resolve({
                  'main.css': 'main.css',
                  'main.js': 'main.js',
                });
              }
              if (manifest === 'vendors') {
                res = Promise.resolve({
                  'vendors.js': 'vendors.js',
                });
              }
              return res;
            },
          },
        });

        const { getAssets } = require('../get-assets'); // eslint-disable-line global-require
        const result = await getAssets({
          host: 'demo.reva.tech',
          query: { min: 'false', dMode: 'true' },
          useDevMode: false,
          jsManifests: ['main', 'vendors'],
          cssFiles: ['main.css'],
          jsFiles: ['vendors.js', 'main.js'],
        });
        expect(result).toMatchSnapshot();
      });
    });

    describe('and host is a local host ', () => {
      it('should return omit the useLocalAssets and return the local assets ', async () => {
        mockModules({
          '../../helpers/xfs': {
            readJSON: manifest => {
              let res;
              if (manifest === 'main') {
                res = Promise.resolve({
                  'main.css': 'main.css',
                  'main.js': 'main.js',
                });
              }
              if (manifest === 'vendors') {
                res = Promise.resolve({
                  'vendors.js': 'vendors.js',
                });
              }
              return res;
            },
          },
        });

        const { getAssets } = require('../get-assets'); // eslint-disable-line global-require
        const result = await getAssets({
          host: 'demo.local.env.reva.tech',
          query: { dMode: 'true' },
          useDevMode: false,
          jsManifests: ['main', 'vendors'],
          cssFiles: ['main.css'],
          jsFiles: ['vendors.js', 'main.js'],
        });
        expect(result).toMatchSnapshot();
      });
    });
  });

  describe('when NODE_ENV = development and dMode = true ', () => {
    it('should omit the useLocalAssets and return the local assets ', async () => {
      mockModules({
        '../../helpers/xfs': {
          readJSON: manifest => {
            let res;
            if (manifest === 'main') {
              res = Promise.resolve({
                'main.css': 'main.css',
                'main.js': 'main.js',
              });
            }
            if (manifest === 'vendors') {
              res = Promise.resolve({
                'vendors.js': 'vendors.js',
              });
            }
            return res;
          },
        },
      });

      const { getAssets } = require('../get-assets'); // eslint-disable-line global-require
      const result = await getAssets({
        host: 'demo.local.env.reva.tech',
        query: { dMode: 'true' },
        useDevMode: true,
        jsManifests: ['main', 'vendors'],
        cssFiles: ['main.css'],
        jsFiles: ['vendors.js', 'main.js'],
      });
      expect(result).toMatchSnapshot();
    });
  });
});
