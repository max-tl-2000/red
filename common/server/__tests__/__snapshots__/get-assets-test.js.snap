// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-assets-test when NODE_ENV = development and dMode = true  should omit the useLocalAssets and return the local assets  1`] = `
Object {
  "assetsHostname": "static.local.env.reva.tech",
  "cssAssets": Array [
    "//static.local.env.reva.tech/libs/font/roboto/roboto.css",
    "//static.local.env.reva.tech/libs/fullcalendar/fullcalendar_1494301536.min.css",
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.css",
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.css",
    "//static.local.env.reva.tech/dist/main.css",
  ],
  "getResource": [Function],
  "getStaticResource": [Function],
  "jsAssets": Array [
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.min.js",
    "//static.local.env.reva.tech/libs/jquery/jquery_1494301536.js",
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.js",
    "//static.local.env.reva.tech/libs/moment/moment_1494301536.min.js",
    "//static.local.env.reva.tech/libs/fullcalendar/fullcalendar_1494301536.min.js",
    "//static.local.env.reva.tech/dist/vendors.js",
    "//static.local.env.reva.tech/dist/main.js",
  ],
  "useSelfHostedAssets": true,
}
`;

exports[`get-assets-test when NODE_ENV = production and query.dMode = true and host is a local host  should return omit the useLocalAssets and return the local assets  1`] = `
Object {
  "assetsHostname": "static.local.env.reva.tech",
  "cssAssets": Array [
    "//fonts.googleapis.com/css?family=Roboto:300,400,500",
    "//cdnjs.cloudflare.com/ajax/libs/fullcalendar/3.1.0/fullcalendar.min.css",
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.css",
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.css",
    "//static.local.env.reva.tech/dist/main.min.css",
  ],
  "getResource": [Function],
  "getStaticResource": [Function],
  "jsAssets": Array [
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.min.js",
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/jquery/3.0.0/jquery.min.js",
    },
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.js",
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/moment.js/2.17.1/moment.min.js",
    },
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/fullcalendar/3.1.0/fullcalendar.min.js",
    },
    "//static.local.env.reva.tech/dist/vendors.min.js",
    "//static.local.env.reva.tech/dist/main.min.js",
  ],
  "useSelfHostedAssets": false,
}
`;

exports[`get-assets-test when NODE_ENV = production and query.dMode = true and host is production host and query.min = false  should return the assets from local and should return no min extensions for main.js and vendors.js and main.css and vendor.css  1`] = `
Object {
  "assetsHostname": "static.local.reva.tech",
  "cssAssets": Array [
    "//fonts.googleapis.com/css?family=Roboto:300,400,500",
    "//cdnjs.cloudflare.com/ajax/libs/fullcalendar/3.1.0/fullcalendar.min.css",
    "//static.local.reva.tech/libs/materialize/materialize_1494301537.min.css",
    "//static.local.reva.tech/libs/materialize-nouislider/nouislider_1494301536.css",
    "//static.local.reva.tech/dist/main.css",
  ],
  "getResource": [Function],
  "getStaticResource": [Function],
  "jsAssets": Array [
    "//static.local.reva.tech/libs/materialize-nouislider/nouislider_1494301536.min.js",
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/jquery/3.0.0/jquery.min.js",
    },
    "//static.local.reva.tech/libs/materialize/materialize_1494301537.min.js",
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/moment.js/2.17.1/moment.min.js",
    },
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/fullcalendar/3.1.0/fullcalendar.min.js",
    },
    "//static.local.reva.tech/dist/vendors.js",
    "//static.local.reva.tech/dist/main.js",
  ],
  "useSelfHostedAssets": false,
}
`;

exports[`get-assets-test when NODE_ENV = production and query.dMode = true and host is production host should return the assets from local 1`] = `
Object {
  "assetsHostname": "static.local.reva.tech",
  "cssAssets": Array [
    "//fonts.googleapis.com/css?family=Roboto:300,400,500",
    "//cdnjs.cloudflare.com/ajax/libs/fullcalendar/3.1.0/fullcalendar.min.css",
    "//static.local.reva.tech/libs/materialize/materialize_1494301537.min.css",
    "//static.local.reva.tech/libs/materialize-nouislider/nouislider_1494301536.css",
    "//static.local.reva.tech/dist/main.min.css",
  ],
  "getResource": [Function],
  "getStaticResource": [Function],
  "jsAssets": Array [
    "//static.local.reva.tech/libs/materialize-nouislider/nouislider_1494301536.min.js",
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/jquery/3.0.0/jquery.min.js",
    },
    "//static.local.reva.tech/libs/materialize/materialize_1494301537.min.js",
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/moment.js/2.17.1/moment.min.js",
    },
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/fullcalendar/3.1.0/fullcalendar.min.js",
    },
    "//static.local.reva.tech/dist/vendors.min.js",
    "//static.local.reva.tech/dist/main.min.js",
  ],
  "useSelfHostedAssets": false,
}
`;

exports[`get-assets-test when NODE_ENV = production and query.local = true should return the list of files from cdn ignoring local=true, min extensions for main.js and vendors.js and main.css 1`] = `
Object {
  "assetsHostname": "static.local.env.reva.tech",
  "cssAssets": Array [
    "//fonts.googleapis.com/css?family=Roboto:300,400,500",
    "//cdnjs.cloudflare.com/ajax/libs/fullcalendar/3.1.0/fullcalendar.min.css",
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.css",
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.css",
    "//static.local.env.reva.tech/dist/main.min.css",
  ],
  "getResource": [Function],
  "getStaticResource": [Function],
  "jsAssets": Array [
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.min.js",
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/jquery/3.0.0/jquery.min.js",
    },
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.js",
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/moment.js/2.17.1/moment.min.js",
    },
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/fullcalendar/3.1.0/fullcalendar.min.js",
    },
    "//static.local.env.reva.tech/dist/vendors.min.js",
    "//static.local.env.reva.tech/dist/main.min.js",
  ],
  "useSelfHostedAssets": false,
}
`;

exports[`get-assets-test when NODE_ENV = production and query.local = undefined should return the list of files from cdn, min extensions for main.js and vendors.js and main.css and vendor.css 1`] = `
Object {
  "assetsHostname": "static.local.env.reva.tech",
  "cssAssets": Array [
    "//fonts.googleapis.com/css?family=Roboto:300,400,500",
    "//cdnjs.cloudflare.com/ajax/libs/fullcalendar/3.1.0/fullcalendar.min.css",
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.css",
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.css",
    "//static.local.env.reva.tech/dist/main.min.css",
  ],
  "getResource": [Function],
  "getStaticResource": [Function],
  "jsAssets": Array [
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.min.js",
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/jquery/3.0.0/jquery.min.js",
    },
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.js",
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/moment.js/2.17.1/moment.min.js",
    },
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/fullcalendar/3.1.0/fullcalendar.min.js",
    },
    "//static.local.env.reva.tech/dist/vendors.min.js",
    "//static.local.env.reva.tech/dist/main.min.js",
  ],
  "useSelfHostedAssets": false,
}
`;

exports[`get-assets-test when NODE_ENV = production and query.min = false should return the list of files from cdn, no min extensions for main.js and vendors.js and main.css and vendor.css 1`] = `
Object {
  "assetsHostname": "static.local.env.reva.tech",
  "cssAssets": Array [
    "//fonts.googleapis.com/css?family=Roboto:300,400,500",
    "//cdnjs.cloudflare.com/ajax/libs/fullcalendar/3.1.0/fullcalendar.min.css",
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.css",
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.css",
    "//static.local.env.reva.tech/dist/main.css",
  ],
  "getResource": [Function],
  "getStaticResource": [Function],
  "jsAssets": Array [
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.min.js",
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/jquery/3.0.0/jquery.min.js",
    },
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.js",
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/moment.js/2.17.1/moment.min.js",
    },
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/fullcalendar/3.1.0/fullcalendar.min.js",
    },
    "//static.local.env.reva.tech/dist/vendors.js",
    "//static.local.env.reva.tech/dist/main.js",
  ],
  "useSelfHostedAssets": false,
}
`;

exports[`get-assets-test when useDevMode = false and one of them contain the local flag should return the list of files from local and no min extensions for main.js and vendors.js 1`] = `
Object {
  "assetsHostname": "static.local.env.reva.tech",
  "cssAssets": Array [
    "//fonts.googleapis.com/css?family=Roboto:300,400,500",
    "//cdnjs.cloudflare.com/ajax/libs/fullcalendar/3.1.0/fullcalendar.min.css",
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.css",
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.css",
    "/dist/main.min.css",
  ],
  "getResource": [Function],
  "getStaticResource": [Function],
  "jsAssets": Array [
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.min.js",
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/jquery/3.0.0/jquery.min.js",
    },
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.js",
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/moment.js/2.17.1/moment.min.js",
    },
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/fullcalendar/3.1.0/fullcalendar.min.js",
    },
    "//static.local.env.reva.tech/dist/vendors.min.js",
    "/dist/main.min.js",
  ],
  "useSelfHostedAssets": false,
}
`;

exports[`get-assets-test when useDevMode = true and one of them contain the local flag should return the list of files from local and min extensions for main.js and vendors.js 1`] = `
Object {
  "assetsHostname": "static.local.env.reva.tech",
  "cssAssets": Array [
    "//fonts.googleapis.com/css?family=Roboto:300,400,500",
    "//cdnjs.cloudflare.com/ajax/libs/fullcalendar/3.1.0/fullcalendar.min.css",
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.css",
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.css",
    "/dist/main.min.css",
  ],
  "getResource": [Function],
  "getStaticResource": [Function],
  "jsAssets": Array [
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.min.js",
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/jquery/3.0.0/jquery.min.js",
    },
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.js",
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/moment.js/2.17.1/moment.min.js",
    },
    Object {
      "crossDomain": undefined,
      "src": "//cdnjs.cloudflare.com/ajax/libs/fullcalendar/3.1.0/fullcalendar.min.js",
    },
    "//static.local.env.reva.tech/dist/vendors.min.js",
    "/dist/main.min.js",
  ],
  "useSelfHostedAssets": false,
}
`;

exports[`get-assets-test when useDevMode = true and query.local = false should return the list of files from cdn, no min extensions for main.js and vendors.js and no main.css 1`] = `
Object {
  "assetsHostname": "static.local.env.reva.tech",
  "cssAssets": Array [
    "//static.local.env.reva.tech/libs/font/roboto/roboto.css",
    "//static.local.env.reva.tech/libs/fullcalendar/fullcalendar_1494301536.min.css",
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.css",
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.css",
    "//static.local.env.reva.tech/dist/main.css",
  ],
  "getResource": [Function],
  "getStaticResource": [Function],
  "jsAssets": Array [
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.min.js",
    "//static.local.env.reva.tech/libs/jquery/jquery_1494301536.js",
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.js",
    "//static.local.env.reva.tech/libs/moment/moment_1494301536.min.js",
    "//static.local.env.reva.tech/libs/fullcalendar/fullcalendar_1494301536.min.js",
    "//static.local.env.reva.tech/dist/vendors.js",
    "//static.local.env.reva.tech/dist/main.js",
  ],
  "useSelfHostedAssets": true,
}
`;

exports[`get-assets-test when useDevMode = true and query.local = undefined should return the list of files from local and no min extensions for main.js and vendors.js 1`] = `
Object {
  "assetsHostname": "static.local.env.reva.tech",
  "cssAssets": Array [
    "//static.local.env.reva.tech/libs/font/roboto/roboto.css",
    "//static.local.env.reva.tech/libs/fullcalendar/fullcalendar_1494301536.min.css",
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.css",
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.css",
    "//static.local.env.reva.tech/dist/main.css",
  ],
  "getResource": [Function],
  "getStaticResource": [Function],
  "jsAssets": Array [
    "//static.local.env.reva.tech/libs/materialize-nouislider/nouislider_1494301536.min.js",
    "//static.local.env.reva.tech/libs/jquery/jquery_1494301536.js",
    "//static.local.env.reva.tech/libs/materialize/materialize_1494301537.min.js",
    "//static.local.env.reva.tech/libs/moment/moment_1494301536.min.js",
    "//static.local.env.reva.tech/libs/fullcalendar/fullcalendar_1494301536.min.js",
    "//static.local.env.reva.tech/dist/vendors.js",
    "//static.local.env.reva.tech/dist/main.js",
  ],
  "useSelfHostedAssets": true,
}
`;
